src/backend/access/transam/xlog.c

```diff
@@ -6688,6 +6688,10 @@ StartupXLOG(void)
 		 */
 		if (access(BACKUP_LABEL_FILE, F_OK) != 0)
 				SyncAllXLogFiles();
+				
+		Assert(shmCleanupBackends != NULL);
+		*shmCleanupBackends = true;
```

Comment:
This has not been here before. Do we need it?

Reply:
FYI: https://github.com/arenadata/gpdb/pull/1065#discussion_r1783907736

Reply:
I looked at it and I don't really see the meaning of this code. In any case, we will set this variable to true at the end of the function. Is there a situation when we exit this function earlier (except for errors)? Could we need this value earlier?

Reply:
There are errors possible, of different severity, this is the way to exit. Called functions are also capable to throw an error. In this case we might want to exit with this flag set, which means that MPP backends should be terminated. IMHO it's better to have this flag as in 7.x than not. @RekGRpth , what do you think?

Reply:
I think to prove the need for this change - we need to come up with a scenario when it is needed. Like in the [patch](https://github.com/arenadata/gpdb/pull/567), a scenario was come up for a similar change at the end of the function.

Reply:
Okay, let's consider that if this change is needed it's a matter of a different issue/PR.